subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/SnapshotUtil.java

```diff
@@ -193,9 +250,21 @@ private static boolean isRegularFileOrDirectory(MetadataSnapshot metadataSnapsho
     }
 
     private interface DescendantHandler<T> {
```

Comment:
The type name is `DescendantHandler` but it seems to handle things other than descendants, no? Is it a `NodeHandler` perhaps?

Reply:
I renamed it to `PathRelationshipHandler` and the method to `handlePathRelationship`, since this is about the relationship of two relative paths.